src/coreclr/src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -356,6 +356,12 @@ internal static RuntimeAssembly InternalLoad(AssemblyName assemblyName, ref Stac
         // given name.  (Name should not include path.)
         public override FileStream? GetFile(string name)
         {
+            if (Location == "")
+            {
+                // Throw if the assembly was loaded from memory, indicated by Location returning an empty string
+                throw new InvalidOperationException(SR.InvalidOperation_NoFileTableInInMemoryAssemblies);
```

Comment:
Alternatively we could throw `FileNotFoundException` - it's not as good as `InvalidOperationException` but it's "close enough" and it is much less likely to be a breaking change (since `FileNotFoundException` derives from `IOException`, so only direct type equality checks would be broken). The specific exception message should be enough really.

I think the higher order reason for this fix is to make the behavior stable/predictable and not "by accident". Having a nice exception is just a bonus.